### PR TITLE
feat: add pose stamped to psim

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Related Issue(required)

#222

## Description(required)

add or remove pose stamped topic from psim

candidate
- /localization/pose_estimator/debug_pose
- /localization/pose_estimator/fake_pose
- /simulation/simple_planning_simulator/fake_pose

## Review Procedure(required)

see node info current_pose
```
/simulation/simple_planning_simulator
  Subscribers:
    /control/command/control_cmd: autoware_auto_control_msgs/msg/AckermannControlCommand
    /control/command/gear_cmd: autoware_auto_vehicle_msgs/msg/GearCommand
    /control/command/hazard_lights_cmd: autoware_auto_vehicle_msgs/msg/HazardLightsCommand
    /control/command/turn_indicators_cmd: autoware_auto_vehicle_msgs/msg/TurnIndicatorsCommand
    /initialpose: geometry_msgs/msg/PoseWithCovarianceStamped
    /parameter_events: rcl_interfaces/msg/ParameterEvent
    /planning/scenario_planning/trajectory: autoware_auto_planning_msgs/msg/Trajectory
    /tf: tf2_msgs/msg/TFMessage
    /tf_static: tf2_msgs/msg/TFMessage
  Publishers:
    /localization/pose_estimator/fake_pose geometry_msgs/msg/PoseStamped
    /localization/kinematic_state: nav_msgs/msg/Odometry
....
  Service Clients:

  Action Servers:

  Action Clients:

```

## Related PR(optional)

this can also be tested with pose2twist

```
ros2 launch pose2twist pose2twist.launch.xml input_pose_topic:=/curernt_pose
```

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@yukkysaito @kenji-miyake cc: @TakaHoribe
I mentioned this pr in .auto sync and approved to add.
I choose `/localization/pose_estimator/debug_pose`  instead of ` /curent_pose`

**Comment by kenji-miyake** [comment]
@taikitanaka3 Could you explain why `/localization/kinematic_state` isn't enough?
(Anyway, `/current_pose` should be removed.)

**Comment by taikitanaka3** [comment]
@kenji-miyake 
pose2twist package assumes `pose stamped` topic from ndt scan matcher and there is no package that use pose topic from `/localization/kinematic_state` so it's not enough to it will be better to add `odom2twist` package or add pose publisher node. 
https://github.com/autowarefoundation/autoware.universe/blob/tier4/proposal/localization/pose2twist/launch/pose2twist.launch.xml

File: simulator/simple_planning_simulator/launch/simple_planning_simulator.launch.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ def launch_setup(context, *args, **kwargs):
             ('input/turn_indicators_command', '/control/command/turn_indicators_cmd'),
             ('input/hazard_lights_command', '/control/command/hazard_lights_cmd'),
             ('input/trajectory', '/planning/scenario_planning/trajectory'),
+            ('output/pose', '/localization/pose_estimator/debug_pose'),
```
Comment (taikitanaka3) [inline simulator/simple_planning_simulator/launch/simple_planning_simulator.launch.py:65]:
```suggestion
            ('output/pose', '/localization/pose_estimator/fake_pose'),
```

**Comment by kenji-miyake** [comment]
For what use case do you need to run `pose2twist` in the planning simulation? :thinking:

**Comment by taikitanaka3** [comment]
@kenji-miyake 
when testing calibration tools like steering offset estimation which needs ndt's pose because ekf needs parameter tuning
or testing pose2twitst node with simulation. so in case mocking ndt pose out put is necessary this topic is necessary.

**Comment by kenji-miyake** [comment]
> when testing calibration tools like steering offset estimation which needs ndt's pose

Please link the design document of the calibration tool?

> testing pose2twitst node with simulation

I believe this should be done with integration tests like `launch_testing`.

**Comment by taikitanaka3** [comment]
@kenji-miyake 
> Please link the design document of the calibration tool?
Ok I'm going to create it later.

**Comment by taikitanaka3** [comment]
close this now for low priority

**Pull request closed without merging**